build_for_portal.py

```diff
@@ -442,14 +442,17 @@ def reformat_for_drupal(info):
 
     # Reformat the data
     for book in books:
-        log.info("Processing %s", book["Dir"])
-        book_src_dir = os.path.join(src_dir, book["Dir"])
+        try:
+            log.info("Processing %s", book['Dir'])
+            book_src_dir = os.path.join(src_dir, book['Dir'])
 
-        if info["all_in_one"]:
-            images_dir = os.path.join(dest_dir, "images")
-        else:
-            book_dest_dir = os.path.join(dest_dir, book["Dir"])
-            images_dir = os.path.join(book_dest_dir, "images")
+            if info['all_in_one']:
+                images_dir = os.path.join(dest_dir, "images")
+            else:
+                book_dest_dir = os.path.join(dest_dir, book['Dir'])
+                images_dir = os.path.join(book_dest_dir, "images")
+        except Exception as e:
+            logging.exception("%s", e)
```

Comment:
I don't see a possible exception here except if the book node does not have a Dir entry. 

The book nodes are all the top level nodes of the topic map file that  have the current distro in "Distros".

So, are you saying here that in case such a node does not have a "Dir" entry, the exception should be logged but the processing should continue?

Without this modification, the exception will crash out the script, thus failing the CI. I thought this was the right approach as all nodes should have the Dir entry?

Reply:
Think you are right. I'll revert this.